arch/sim/src/sim/win/sim_hostfs.c

```diff
@@ -120,6 +120,11 @@ int host_open(const char *pathname, int flags, nuttx_mode_t mode)
       mapflags |= O_TRUNC;
     }
 
+  if (flags & NUTTX_O_BINARY)
+    {
+      mapflags |= O_BINARY;
+    }
```

Comment:
* have you tested this? the definition of `NUTTX_O_BINARY` has been broken since https://github.com/apache/nuttx/pull/6357 it seems.
* why don't you always use O_BINARY for the host? afaik nuttx doesn't really have text mode.

Reply:
It looks like our internal branch doesn't have this PR https://github.com/apache/nuttx/pull/6357, Could we bring back O_BINARY support?

Reply:
@yamt  I bring back the O_BINARY flag to support covert the default translate mode:
https://github.com/apache/nuttx/pull/9000

Reply:
> the semantics of nuttx (and many of unix-like systems) is "always O_BINARY". it seems a bit awkward to distinguish binary/text only for hostfs. isn't it simpler to always use O_BINARY in hostfs for windows?

we have some windows developers hope that the hosfs could be support both un/translated, I think bring back the O_BINARY define is a simply resolution for this case...

Reply:
@yamt  think twice, In order to unify the translation behavior with unix:
1. set O_BINARY for hostfs as default
2. enable default text mode if the application specifies flag O_TEXT

Could you please review again?

Reply:
> > the semantics of nuttx (and many of unix-like systems) is "always O_BINARY". it seems a bit awkward to distinguish binary/text only for hostfs. isn't it simpler to always use O_BINARY in hostfs for windows?
> 
> we have some windows developers hope that the hosfs could be support both un/translated, I think bring back the O_BINARY define is a simply resolution for this case...

what apps are they using?
usually apps for nuttx (or unix in general) doesn't specify their text/binary intention.
iirc posix even states fopen "b" mode is no-op.

i guess an alternative is to have a mount flag of hostfs.

Reply:
> > > the semantics of nuttx (and many of unix-like systems) is "always O_BINARY". it seems a bit awkward to distinguish binary/text only for hostfs. isn't it simpler to always use O_BINARY in hostfs for windows?
> > 
> > 
> > we have some windows developers hope that the hosfs could be support both un/translated, I think bring back the O_BINARY define is a simply resolution for this case...
> 
> what apps are they using? usually apps for nuttx (or unix in general) doesn't specify their text/binary intention. iirc posix even states fopen "b" mode is no-op.
> 

If you don't pass O_BINARY on Windows, Windows will do the \n<->\r\n. By default, Windows do \n <-> \r\n converson, but Linux/macOS not. We need shield OS difference to keep the consistent behaviour.

> i guess an alternative is to have a mount flag of hostfs.

Option isn't a good choice here. If you open a file with O_BINARY regardless the simulator run on Linux/macOS/Windows, VFS shouldn't do any conversion.

Reply:
> > > > the semantics of nuttx (and many of unix-like systems) is "always O_BINARY". it seems a bit awkward to distinguish binary/text only for hostfs. isn't it simpler to always use O_BINARY in hostfs for windows?
> > > 
> > > 
> > > we have some windows developers hope that the hosfs could be support both un/translated, I think bring back the O_BINARY define is a simply resolution for this case...
> > 
> > 
> > what apps are they using? usually apps for nuttx (or unix in general) doesn't specify their text/binary intention. iirc posix even states fopen "b" mode is no-op.
> 
> If you don't pass O_BINARY on Windows, Windows will do the \n<->\r\n. By default, Windows do \n <-> \r\n converson, but Linux/macOS not. We need shield OS difference to keep the consistent behaviour.

my point is that apps for nuttx usually don't specify binary/text.
thus, regardless of this change, they might not work as you expected.

> 
> > i guess an alternative is to have a mount flag of hostfs.
> 
> Option isn't a good choice here. If you open a file with O_BINARY regardless the simulator run on Linux/macOS/Windows, VFS shouldn't do any conversion.

Reply:
Just ordinary developers, their habits are accumulated on the windows platform, now this PR changed the default mode to O_BINARY, I thinks this meets your suggestion